lib/kube/proxy/roundtrip.go

```diff
@@ -152,9 +157,12 @@ func (s *SpdyRoundTripper) dial(url *url.URL) (net.Conn, error) {
 // connection.
 func (s *SpdyRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	header := utilnet.CloneHeader(req.Header)
-	header.Add(httpstream.HeaderConnection, httpstream.HeaderUpgrade)
-	header.Add(httpstream.HeaderUpgrade, streamspdy.HeaderSpdy31)
-
+	// copyImpersonationHeaders copies the headers from the original request to the new
+	// request headers. This is necessary to forward the original user's impersonation
+	// when multiple kubernetes_users are available.
+	copyImpersonationHeaders(header, s.originalHeaders)
```

Comment:
Can we cover this in UT ?

Reply:
Added in [f83a1eb](https://github.com/gravitational/teleport/pull/21102/commits/f83a1eb9fcfa363c5726fe7c627a891a7bb367e6)